TicTacToe/TicTacToe.cs

```diff
@@ -27,61 +28,176 @@ public static void Main()
 
         public static void GetInput()
         {
+            //Ternary Operator for taking turns
+            playerTurn = (playerTurn == "X") ? "O" : "X";
+
             Console.WriteLine("Player " + playerTurn);
             Console.WriteLine("Enter Row:");
-            int row = int.Parse(Console.ReadLine());
-            Console.WriteLine("Enter Column:");
-            int column = int.Parse(Console.ReadLine());
+            int row = 0;
+            int column = 0;
+            //Try function in case of invalid input
+            try
+            {
+                row = Convert.ToInt32((Console.ReadLine()));
+                //Test for only 0, 1, or 2 inputs for Row
+                if (test1(row))
+                {
+                    Console.WriteLine("Enter Column:");
+
+                    column = Convert.ToInt32((Console.ReadLine()));
+                    
+                    //Test for only 0, 1, or 2 inputs for Column
+                    if (test2(column))
+                    {
+                        //if Both test1 and test2 run through, PlaceMark will be set
+                        PlaceMark(row, column);
+                    }
+                    else
+                    {
+                        Console.WriteLine("Invalid placement or Column! Please enter 0, 1, or 2 only.");
+                        if (playerTurn == "X")
+                        {
+                            playerTurn = "O";
+                        }
+                        else playerTurn = "X";
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Invalid placement for Row! Please enter 0, 1, or 2 only.");
+                    if (playerTurn == "X")
+                    {
+                        playerTurn = "O";
+                    }
+                    else playerTurn = "X";
+                }
+            }
+            //catch function to notify user of bad input
+            catch
+            {
+                //resets the playerTurn
+                if (playerTurn == "X")
+                {
+                    playerTurn = "O";
+                }
+                else playerTurn = "X";
+                Console.WriteLine("Invalid placement! Please enter 0, 1, or 2 only.");
```

Comment:
Good job! I love that you are catching user input errors, but this section of code does not run if the user enters 3 for instance. `catch` blocks only run when an exception is throw inside the `try` block.